File: web/src/api/browserHelper.ts
/**
 * 创建和读取浏览器助手配对会话；页面拿到受限 token 后再通过扩展 bridge 发起采集。
 */
import { client } from "./client";

export interface BrowserHelperProfileSessionOption {
  session_id: string;
  label: string;
  is_default: boolean;
}

export interface BrowserHelperSession {
  access_token: string;
  token_type: "bearer";
  expires_at: string;
  profile_sessions: BrowserHelperProfileSessionOption[];
}

export async function createBrowserHelperSession(): Promise<BrowserHelperSession> {
  const response = await client.post<BrowserHelperSession>("/api/v1/browser-helper/sessions");
  return response.data;
}

File: app/schemas/browser_helper.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class BrowserHelperProfileSessionOption(BaseModel):
    session_id: str
    resume_profile_id: str | None = None
    label: str
    is_default: bool = False


class BrowserHelperSavedJobOption(BaseModel):
    saved_job_id: str
    title: str
    company: str | None = None


class BrowserHelperContextCatalog(BaseModel):
    saved_jobs: list[BrowserHelperSavedJobOption] = Field(default_factory=list)


class BrowserHelperSessionCreateResponse(BaseModel):
    access_token: str
    token_type: str = "bearer"
    expires_at: datetime
    profile_sessions: list[BrowserHelperProfileSessionOption] = Field(default_factory=list)
